selenium/src/test/java/au/org/emii/portal/utils/SeleniumUtil.java

```diff
@@ -0,0 +1,135 @@
+                System.out.println(curlResponse);
+            }
+        } catch (IOException e) {
+            log.error("Unable to execute curl command", e);
+        }
+        return curlResponse;
+    }
+
+    //Find Broken / Invalid Images on a Page
+    public static int validateInvalidImages(WebDriver driver) {
+        int invalidImageCount = 0;
+        List<WebElement> imagesList = driver.findElements(By.tagName("img"));
+        List<String> uniqueSrcList = new ArrayList<>();
+        log.info("Total no. of images are " + imagesList.size());
+        for (WebElement imgElement : imagesList) {
+            if (imgElement != null && imgElement.getAttribute("src") != null && !uniqueSrcList.contains(imgElement.getAttribute("src"))) {
+                uniqueSrcList.add(imgElement.getAttribute("src"));
+                invalidImageCount = verifyURLStatus(imgElement.getAttribute("src"), invalidImageCount);
+            }
+        }
+        log.info("Total no. of invalid images are " + invalidImageCount);
+        return invalidImageCount;
+    }
+
+    //Find out broken links on website
+    public static int validateInvalidLinks(WebDriver driver) {
+
+        int invalidLinksCount = 0;
+        List<WebElement> anchorTagsList = driver.findElements(By
+                .tagName("a"));
+        List<String> uniqueAnchorList = new ArrayList<>();
+        log.info("Total no. of links are "
+                + anchorTagsList.size());
+
+        for (int i = 0; i < anchorTagsList.size(); i++) {
+            WebElement anchorTag = anchorTagsList.get(i);
+            String url = anchorTag.getAttribute("href");
+            if (anchorTag != null && url != null && !uniqueAnchorList.contains(url)) {
+                uniqueAnchorList.add(url);
+                if (!url.contains("javascript")) {
+                    invalidLinksCount = verifyURLStatus(url, invalidLinksCount);
+                } else {
+                    log.error("Invalid link " + url);
+                    invalidLinksCount++;
+                }
+            }
+        }
+
+        log.info("Total no. of invalid links are "
+                + invalidLinksCount);
+        return invalidLinksCount;
+    }
+
+    public static int verifyURLStatus(String URL, int invalidLinksCount) {
+        log.info("Validating URL " + URL);
+        //Validating the contact us email
+        if (URL.startsWith("mailto:info@aodn.org.au?subject=")) {
+            return invalidLinksCount;
+        }
+        try {
+            SSLContext sslContext = SSLContexts.custom()
+                    .loadTrustMaterial(null, new TrustStrategy() {
+
+                        @Override
+                        public boolean isTrusted(final X509Certificate[] chain, final String authType) throws CertificateException {
+                            return true;
+                        }
+                    })
+                    .useTLS()
+                    .build();
```

Comment:
The methods used here ar deprecated - perhaps use undeprecated alternatives instead.